src/test/java/com/commercetools/sync/products/utils/ProductSyncUtilsTest.java

```diff
@@ -148,16 +145,79 @@ public void buildActions_FromDraftsWithMultipleDifferentValues_ShouldBuildUpdate
             AddExternalImage.ofVariantId(1,
                 Image.of("https://53346cfbf3c7e017ed3d-6de74c3efa80f1c837c6a988b57abe66.ssl.cf3.rackcdn.com/"
                     + "new-image-2.png", ImageDimensions.of(0, 0), null), true),
-            SetPrices.of(1,
-                asList(PriceDraft.of(BigDecimal.valueOf(1.18), DefaultCurrencyUnits.EUR)
-                                 .withChannel(Channel.referenceOfId("channel-key_1")),
-                    PriceDraft.of(BigDecimal.valueOf(1), MoneyImpl.createCurrencyByCode("EGP"))
-                              .withChannel(Channel.referenceOfId("channel-key_1")))),
+            RemovePrice.of(Price.of(MoneyImpl.ofCents(108, EUR))
+                                .withId("4dfc8bea-84f2-45bc-b3c2-cdc94bf96f1f"), true),
+            RemovePrice.of(Price.of(MoneyImpl.ofCents(10, createCurrencyByCode("EGP")))
+                                .withId("4dfc8bea-84f2-45bc-b3c2-r9e7wv99vfb"), true),
+            AddPrice.ofVariantId(1,
+                PriceDraft.of(MoneyImpl.ofCents(118, EUR))
+                          .withChannel(Channel.referenceOfId("channel-key_1")), true),
+            AddPrice.ofVariantId(1,
+                PriceDraft.of(MoneyImpl.ofCents(100, createCurrencyByCode("EGP")))
+                          .withChannel(Channel.referenceOfId("channel-key_1")), true),
             AddAsset.ofVariantId(1, assetDraft).withPosition(0).withStaged(true),
             SetSku.of(1, "3065831", true)
         );
     }
 
+    @Test
+    public void buildActions_FromDraftsWithMultipleDifferentValues_ShouldBuildUpdateActionsInCorrectOrder() {
```

Comment:
wouldn't be better to just test that you are calling the right functions with the proper parameters (basically test the behavior)? I mean, product update actions are massive, so trying to test in the same unit test different kind of update actions makes the test huge, the setup gets complicated, and finally difficult to follow and maintain (it's brittle to small changes).

Since you have unit tests for the different types of update actions within the product ("general" update actions such as "Change Name", "Change Slug", etc..., variant update actions, variant images update actions, etc...), wouldn't be better to test for the ProductSyncTestUtils just the following:

"when I call to `ProductSyncUtils.buildActions` with this new product draft and this product, it should call `my_build_product_update_function` with the same product draft and the same product"

i.e when I call to `ProductSyncUtils.buildActions(oldProduct, newProductDraft, productSyncOptions, new HashMap<>())`, it should call `buildSetDescriptionUpdateAction` with `oldProduct`, `newProductDraft`. 

You already have a test for the "buildSetDescriptionUpdateAction", so testing that you are calling it with the proper parameters is enough.

And the same for the rest of build update actions functions... `buildVariantsUpdateActions`, 

For me this way helps:

- the developer to understand what's going on within the ProductTypeSyncUtils --> tests as documentation. Reading the test as it is, it's almost impossible to know why there has to be a RemoveImage.ofVariantId(1), AddExternalImage.ofVariantId(1), etc.. update actions.
- to easily identify the cause if the test fails because the scope is smaller. In this huge test, if an assert fails it's difficult to identify why. maybe it's been a bad setup with wrong values? maybe an actual error inside one of the multiple functions is calling?

Sorry for the long comment 🙈. But I think that for someone that has to maintain the code, this kind of tests are tough to fix and understand the cause if it fails.

It's just my opinion 🙂. WDYT?

Reply:
I think this is a great idea! :)

Reply:
It only makes sense to check all the actual built update actions in the integration tests.

Reply:
@jortizsao but I would rather create a separate PR for it, not to clutter the main goal of this one. I created an issue for it already: https://github.com/commercetools/commercetools-sync-java/issues/292